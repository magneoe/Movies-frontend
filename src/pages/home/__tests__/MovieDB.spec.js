import React from 'react';

import configureStore from 'redux-mock-store';

import Adapter from 'enzyme-adapter-react-16';
import {expect as expectChai} from 'chai';
import Enzyme, { shallow} from 'enzyme';
import sinon from 'sinon';

import ConnectedMovieDB, { mapStateToProps, MovieDB } from '../MovieDB';

let initialState; 
// pass in any middleware if needed into //configureStore
let mockStore = configureStore();
let store;

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: true });
    store = mockStore(initialState);
});

beforeEach(() => {
    store.clearActions();

    initialState =  {
        movieReducer: {
            loading: false,
            comments: [],
            requestConfig: {
                size: 5,
                direction: 0,
                sort: 'title'
            },
            movieData: {
                content: [
                    {
                        id: 7,
                        title: 'Coming to America',
                    }]
            }
        }
    };
});

describe("MovieDB component", () => {
    describe("MapStateToProps", () => {
        it("Should map state to props correctly", () => {
            
            const expectedProps = {
                requestConfig: initialState.movieReducer.requestConfig,
                movieData: initialState.movieReducer.movieData,
                loading: initialState.movieReducer.loading,
                userRatings: [],
                comments: []
            };
            const mappedProps = mapStateToProps(initialState);

            expect(mappedProps).toEqual(expectedProps);
        });

        it("Should map state to props correctly on empty state", () => {
            const appState = {
            };
            const expectedProps = {
                requestConfig: {},
                movieData: {},
                loading: false,
                userRatings: [],
                comments: []
            };
            const mappedProps = mapStateToProps(appState);

            expect(mappedProps).toEqual(expectedProps);
        });
    });

    describe("The display unit", () => {
        it("Should render successfully and loadMovie on componentDidMount", () => {
            store = mockStore(initialState);

            let spy = sinon.spy();
            const wrapper = shallow(<MovieDB actions={{loadMovies: spy}}/>, {disableLifecycleMethods: false});
            expectChai(spy.callCount).to.equal(1);
            expect(wrapper).toMatchSnapshot();
        });

        it("Should render loading wheel whenever in loading state", () => {
            initialState = {...initialState,
                movieReducer: { loading: true}
            };
            store = mockStore(initialState);
            const wrapper = shallow(<ConnectedMovieDB store={store}/>);
            expectChai(wrapper.dive().find('div.lds-roller')).to.have.length(1);
        });

        it("Should render MovieList when not in loading state", () => {
            initialState = {...initialState,
                movieReducer: { loading: false}
            };
            store = mockStore(initialState);
            const wrapper = shallow(<ConnectedMovieDB store={store}/>);
            expectChai(wrapper.dive().find('MovieList')).to.have.length(1);
        });
    });
});