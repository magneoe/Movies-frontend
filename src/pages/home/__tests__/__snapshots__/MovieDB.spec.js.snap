// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDB component The display unit Should render successfully and loadMovie on componentDidMount 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieDB
    actions={
      Object {
        "loadMovies": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PageInfo
          currentPage={0}
          numberOfElements={0}
          onNext={[Function]}
          onPrev={[Function]}
          totalPages={0}
        />,
        <PageSort
          onChangedSortOption={[Function]}
          sortOptions={
            Array [
              "title",
              "year",
              "duration",
            ]
          }
        />,
        <div
          className="movieListContainer"
        >
          <MovieList
            movies={Array []}
            onClick={[Function]}
          />
        </div>,
        <Connect(MovieModal)
          closeModal={[Function]}
          modalIsOpen={false}
          movieId={null}
        />,
      ],
      "style": Object {
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentPage": 0,
          "numberOfElements": 0,
          "onNext": [Function],
          "onPrev": [Function],
          "totalPages": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "defaultDirection": undefined,
          "defaultSortOption": undefined,
          "onChangedSortOption": [Function],
          "sortOptions": Array [
            "title",
            "year",
            "duration",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MovieList
            movies={Array []}
            onClick={[Function]}
          />,
          "className": "movieListContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "movies": Array [],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeModal": [Function],
          "modalIsOpen": false,
          "movieId": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PageInfo
            currentPage={0}
            numberOfElements={0}
            onNext={[Function]}
            onPrev={[Function]}
            totalPages={0}
          />,
          <PageSort
            onChangedSortOption={[Function]}
            sortOptions={
              Array [
                "title",
                "year",
                "duration",
              ]
            }
          />,
          <div
            className="movieListContainer"
          >
            <MovieList
              movies={Array []}
              onClick={[Function]}
            />
          </div>,
          <Connect(MovieModal)
            closeModal={[Function]}
            modalIsOpen={false}
            movieId={null}
          />,
        ],
        "style": Object {
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPage": 0,
            "numberOfElements": 0,
            "onNext": [Function],
            "onPrev": [Function],
            "totalPages": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "defaultDirection": undefined,
            "defaultSortOption": undefined,
            "onChangedSortOption": [Function],
            "sortOptions": Array [
              "title",
              "year",
              "duration",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MovieList
              movies={Array []}
              onClick={[Function]}
            />,
            "className": "movieListContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "movies": Array [],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeModal": [Function],
            "modalIsOpen": false,
            "movieId": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;
